src/main/java/com/faforever/client/preferences/ui/SettingsController.java

```diff
@@ -255,9 +286,29 @@ private void configureThemeSelection(Preferences preferences) {
     );
   }
 
-  private void configureLanguageSelection() {
-    languageComboBox.setItems(FXCollections.singletonObservableList("English"));
-    languageComboBox.getSelectionModel().select(0);
+  private void configureLanguageSelection(Preferences preferences) {
+    languageComboBox.setItems(FXCollections.observableArrayList(LanguageInfo.getAllDisplayNames()));
+    String language = preferences.getLanguage().getLanguage();
+    languageComboBox.getSelectionModel().select(LanguageInfo.getIndexByLanguageCode(language));
+
+    languageComboBox.setDisable(false);
+    languageComboBox.setOnAction(event -> onLanguageSelected(preferences));
+  }
+
+  private void onLanguageSelected(Preferences preferences) {
+
+    if (languageComboBox.getValue().toString() != preferences.getLanguage().getLanguage()) {
```

Comment:
Instead of having 
```
if(something) {
}
```
we prefer:
```
if(!something) {
  return;
}

do
all
of
this
```
to avoid nesting and indentation.